lib/optimize/SideEffectsFlagPlugin.js

```diff
@@ -117,5 +116,34 @@ class SideEffectsFlagPlugin {
 			});
 		});
 	}
+
+	static moduleHasSideEffects(modulePath, flagValue) {
```

Comment:
This is too much. Please only support boolean, string as glob and array of strings as globs.

Reply:
I agree it is too much.  I was trying to satisfy all the variations in the related issues that came up.  I'll trim it down.